### ISSUE TITLE
BoxFile.getInfo(BoxFile.ALL_FIELDS) throws HTTP-415 for files without preview

### ISSUE THREAD

**Issue by cpuzicha** [box/box-java-sdk#1215]
- [ x] I have checked that the [SDK documentation][sdk-docs] doesn't solve my issue.
- [ x] I have checked that the [API documentation][api-docs] doesn't solve my issue.
- [ x] I have searched the [Box Developer Forums][dev-forums] and my issue isn't already reported (or if it has been reported, I have attached a link to it, for reference).
- [ x] I have searched [Issues in this repo][github-repo] and my issue isn't already reported.

### Description of the Issue
BoxFile.getInfo(BoxFile.ALL_FIELDS) throws a BoxAPIResponseException for a file that cannot be previewed (sample file attached in a ZIP).
This is an unexpected behavior - I am not even sure which fields I am supposed to leave out for this.

Probably the same issue:
https://support.box.com/hc/en-us/community/posts/360052127393-Getting-415-while-file-metadata-of-zip-files

### Steps to Reproduce

1. Upload the sample file from the attached ZIP, note down the file and user id.
2. Create a BoxAPIConnection for that user
3. Execute the following snipped 
```
BoxAPIConnection boxApi = ...
BoxFile file = new BoxFile(boxApi, fileId);
file.getInfo(BoxFile.ALL_FIELDS);
```

5. Observe the stacktrace

### Expected Behavior
Normal JSON response, leaving the fields blank or out that are filled from the preview functionality.

### Error Message, Including Stack Trace
> com.box.sdk.BoxAPIResponseException: The API returned an error code [415 | ...] unsupported_media_type - Previews for 'cer' files are not yet supported.
>	at com.box.sdk.BoxAPIResponse.toBoxResponse(BoxAPIResponse.java:147)
>	at com.box.sdk.BoxAPIRequest.trySend(BoxAPIRequest.java:616)
>	at com.box.sdk.BoxAPIRequest.send(BoxAPIRequest.java:380)
>	at com.box.sdk.BoxJSONRequest.send(BoxJSONRequest.java:106)
>	at com.box.sdk.BoxJSONRequest.send(BoxJSONRequest.java:17)
>	at com.box.sdk.BoxAPIRequest.send(BoxAPIRequest.java:347)
>	at com.box.sdk.BoxJSONRequest.send(BoxJSONRequest.java:101)
>	at com.box.sdk.BoxFile.getInfo(BoxFile.java:453)

### Screenshots
[sampleFile.zip](https://github.com/box/box-java-sdk/files/13439362/sampleFile.zip)

### Versions Used
Java SDK: 4.6.1
Java: AdoptOpen JDK jdk8u275-b01

[sdk-docs]: ./doc
[api-docs]: https://developer.box.com/docs
[dev-forums]: https://community.box.com/t5/Platform-and-Development-Forum/bd-p/DeveloperForum

[github-repo]: https://github.com/box/box-java-sdk/search?type=Issues

**Comment by arjankowski** [comment]
Hi @cpuzicha,

The issue here is with `the expiring_embed_link` field. 
When sending the `expiring_embed_link` in the `fields` query params, 
it creates a public file preview URL for the given file (https://developer.box.com/guides/embed/box-view/create-preview/#generate-an-embed-url-for-the-file).

In the case where the file does not have an extension, Box is unable to successfully generate a file representation and returns a 415 error (https://support.box.com/hc/en-us/articles/360044192513-API-Content-415-unsupported-media-type-).

To resolve the error, you have two options:

1) Remove the value of `expiring_embed_link` in the fields query params.
2) Add an extension to the file causing the error, which allows the generation of a preview URL (https://support.box.com/hc/en-us/articles/360043695794-Viewing-Different-File-Types-Supported-in-Box-Content-Preview).

In summary, the error is not specific to the SDK, as the same error would be received when making a CURL call with the same parameters as BOX API says (https://developer.box.com/reference/get-files-id/#param-415-application/json).

I hope this helps!
Best regards,
Artur

**Comment by cpuzicha** [comment]
Then my issue moves to the REST API - the error message is not helpful at all (but - my fault - your provided documentation is). 
I also consider the SDK behavior unsatisfactory:
1.It requires me to remove things from a constant,
2.Based on information that I cannot acquire programmatically: The list of preview-supported file extensions.

It would be helpful to add a safer constant for "all fields except expiring_embed_link" that just works.
Or provide me with a different constant for file extensions that are supported for preview.

**Comment by arjankowski** [comment]
Hi @cpuzicha,

`BoxFile` is a subclass of `BoxItem`, which also defines the constant `BoxItem.ALL_FIELDS`. 
This constant does **not** include `expiring_embed_link`and it likely excludes several other fields as well. 

Could you please check if this constant meets your requirements?

**Comment by cpuzicha** [comment]
Thanks for the quick reply.

Present in BoxItem but not in BoxFile
* folder_upload_email
* item_collection
* sync_state
* can_non_owners_invite
* expires_at

Present in BoxFile but not in BoxInfo
* watermark_info
* metadata
* representations
* is_external_only
* expiring_embed_link
* allowed_invitee_roles
* disposition_at
* is_accessible_via_shared_link

I think the former list should be empty.
Apart from that: Technically it would probably work.

And just for clarification: I can work around the issue now - it's more about a future improvement of the SDK/REST API.
